mountpoint-s3/tests/fuse_tests/fork_test.rs

```diff
@@ -484,6 +487,186 @@ fn mount_with_sse(
+    Ok(())
+}
+
+#[test]
+fn mount_with_assumed_role_in_other_region() -> Result<(), Box<dyn std::error::Error>> {
+    let (bucket, prefix) = get_test_bucket_and_prefix("mount_with_assumed_role_in_other_region");
+    let subsession_role = get_subsession_iam_role();
+    let region = get_test_region();
+    let other_region = get_non_test_region();
+    let mount_point = assert_fs::TempDir::new()?;
+    let profile_name = "fork_test";
+    let source_profile = "default";
+
+    let config_file = create_cli_config_file(profile_name, source_profile, &subsession_role, &other_region)?;
+
+    let mut cmd = Command::cargo_bin("mount-s3")?;
+    let child = cmd
+        .arg(&bucket)
+        .arg(mount_point.path())
+        .arg(format!("--prefix={prefix}"))
+        .arg("--auto-unmount")
+        .env("AWS_CONFIG_FILE", config_file.path())
+        .env("AWS_PROFILE", profile_name)
+        .env_remove("AWS_ACCESS_KEY_ID")
+        .env_remove("AWS_SECRET_ACCESS_KEY")
+        .env_remove("AWS_SESSION_TOKEN")
+        .spawn()
+        .expect("unable to spawn child");
+
+    let exit_status = wait_for_exit(child);
+
+    // verify mount status and mount entry
+    assert!(exit_status.success());
+    assert!(mount_exists("mountpoint-s3", mount_point.path().to_str().unwrap()));
+
+    test_read_files(&bucket, &prefix, &region, &mount_point.to_path_buf());
+
+    unmount(mount_point.path());
+
+    Ok(())
+}
+
+// Test profile auth without setting a region so it will fallback to global STS endpoint
+#[test]
+fn mount_with_assumed_role_no_region() -> Result<(), Box<dyn std::error::Error>> {
+    let (bucket, prefix) = get_test_bucket_and_prefix("mount_with_assumed_role_no_region");
+    let subsession_role = get_subsession_iam_role();
+    let region = get_test_region();
+    let mount_point = assert_fs::TempDir::new()?;
+    let profile_name = "fork_test";
+    let source_profile = "default";
+
+    let config_file = create_cli_config_file(profile_name, source_profile, &subsession_role, &region)?;
+
+    let mut cmd = Command::cargo_bin("mount-s3")?;
+    let child = cmd
+        .arg(&bucket)
+        .arg(mount_point.path())
+        .arg(format!("--prefix={prefix}"))
+        .arg("--auto-unmount")
+        .env("AWS_CONFIG_FILE", config_file.path())
+        .env("AWS_PROFILE", profile_name)
+        .env_remove("AWS_ACCESS_KEY_ID")
+        .env_remove("AWS_SECRET_ACCESS_KEY")
+        .env_remove("AWS_SESSION_TOKEN")
+        .env("AWS_EC2_METADATA_DISABLED", "true")
+        .env_remove("AWS_REGION")
+        .env_remove("AWS_DEFAULT_REGION")
+        .spawn()
+        .expect("unable to spawn child");
+
+    let exit_status = wait_for_exit(child);
+
+    // verify mount status and mount entry
+    assert!(exit_status.success());
+    assert!(mount_exists("mountpoint-s3", mount_point.path().to_str().unwrap()));
+
+    test_read_files(&bucket, &prefix, &region, &mount_point.to_path_buf());
+
+    unmount(mount_point.path());
+
+    Ok(())
+}
```

Comment:
We are still setting a region in the profile, right? I'm not sure this tests anything right now.

Do we instead want to make region optional when passing to `create_cli_config_file(...)`?

Reply:
yeah, this was a miss from refactoring. It should be optional.